testbed/correctnesstests/utils.go

```diff
@@ -44,6 +45,64 @@ func CreateConfigYaml(
 		}
 	}
 
+	var pipeline1 string
+	switch sender.(type) {
+	case testbed.TraceDataSender:
+		pipeline1 = "traces"
+	case testbed.MetricDataSender:
+		pipeline1 = "metrics"
+	case testbed.LogDataSender:
+		pipeline1 = "logs"
+	default:
+		t.Error("Invalid DataSender type")
+	}
+
+	if connector != nil {
+		pipeline2 := connector.GetReceiverType()
+
+		format := `
+receivers:%v
+exporters:%v
+processors:
+  %s
+
+extensions:
+
+connectors:%v
+
+service:
+  telemetry:
+    metrics:
+      address: 127.0.0.1:%d
+    logs:
+      level: "debug"
+  extensions:
+  pipelines:
+    %s/in:
+      receivers: [%v]
+      processors: [%s]
+      exporters: [%v]
+    %s/out:
+      receivers: [%v]
+      exporters: [%v]
+`
+		return fmt.Sprintf(
+			format,
+			sender.GenConfigYAMLStr(),
+			receiver.GenConfigYAMLStr(),
+			processorsSections,
+			connector.GenConfigYAMLStr(),
+			testutil.GetAvailablePort(t),
+			pipeline1,
+			sender.ProtocolName(),
+			processorsList,
+			connector.ProtocolName(),
+			pipeline2,
+			connector.ProtocolName(),
+			receiver.ProtocolName(),
```

Comment:
I don't understand this one: shouldn't this be an exporter?

Reply:
I believe that in the testbed architecture, the DataReceiver acts as the exporter in the pipeline so that it exports the data to the mocked backend.

Reply:
Ah, I think you are right, yes, thanks for bearing with me :)